lib/jobs/obm-control.js

```diff
@@ -64,18 +62,31 @@ function obmControlJobFactory(
             assert.ok(node, "Node should exist to run OBM command");
             assert.ok(node.obmSettings, "Node should have OBM settings to run OBM command");
 
-            var settings = _.find(node.obmSettings, function(setting) {
-                return setting ? setting.service === self.obmServiceName : false;
-            });
-
-            assert.ok(settings, "Node should have OBM settings for service: " +
-                self.obmServiceName);
+            if (self.obmServiceName) {
+                self.settings = _.find(node.obmSettings, function(setting) {
+                    return setting ? setting.service === self.obmServiceName : false;
+                });
+                // throw error if the exact obm service doesn't exist
+                assert.ok(self.settings, "Node should have OBM settings for service: " +
+                    self.obmServiceName);
+            } else {
+                //find the default obm service
+                if(node.obmSettings.length === 1) {
+                    self.settings = node.obmSettings[0];
+                    logger.info("Defaulting to obmServiceName "+ self.settings.service +
+                        " for node because only one OBM service present.");
+                }else {
+                    assert.fail(true,true, "More than one OBM service assigned to this node," +
+                        " (no default set for which to use)" +
```

Comment:
Also I'd think we have enough information to throw separate error messages depending on whether there are no services assigned, or multiple ones assigned, that way the user doesn't have to guess which one it is.